sessionctx/binloginfo/binloginfo.go

```diff
@@ -49,14 +51,75 @@ type BinlogInfo struct {
 	Client *pumpcli.PumpsClient
 }
 
-// GetPumpsClient gets the pumps client instance.
+// GetPumpsClient get the pumps client instance.
 func GetPumpsClient() *pumpcli.PumpsClient {
 	pumpsClientLock.RLock()
 	client := pumpsClient
 	pumpsClientLock.RUnlock()
 	return client
 }
 
+// GetPumpsClientByLogBin get the pumps client instance by log_bin variable.
+func GetPumpsClientByLogBin() *pumpcli.PumpsClient {
+	var binlogCli *pumpcli.PumpsClient
+	tidbLogBin := variable.GetSysVar(variable.TiDBLogBin)
+	logBin := variable.GetSysVar(variable.LogBin)
+	if (logBin.Scope&variable.ScopeSession == 1 && logBin.Value == "1") ||
```

Comment:
there is a function to judge weather should write binlog, https://github.com/pingcap/tidb/blob/20463d6da90fdf12c0d9d18c15dc33a78334882d/sessionctx/binloginfo/binloginfo.go#L99